src/pocketmine/Player.php

```diff
@@ -1768,7 +1768,7 @@ public function handleDataPacket(DataPacket $packet){
 
 				if($this->isCreative()){
 					$this->inventory->setHeldItemSlot(0);
-				}else{
+				}elseif(!$this->inventory->getHotbarSlotIndex(0) === null){
```

Comment:
This isn't correct anyway. You will convert $this->inventory->getHotbarSlotIndex(0) into a boolean if you do it like this.